Rubberduck.RibbonDispatcher/Sample Ribbons/SampleRibbon/src/RibbonModel.cls

```diff
@@ -0,0 +1,131 @@
+Attribute mCustomButton2.VB_VarHelpID = -1
+Private WithEvents mCustomButton3       As RibbonButton
+Attribute mCustomButton3.VB_VarHelpID = -1
+Private WithEvents mCustomSizeToggle    As RibbonToggleButton
+Attribute mCustomSizeToggle.VB_VarHelpID = -1
+Private WithEvents mCustomOptions       As RibbonDropDown
+Attribute mCustomOptions.VB_VarHelpID = -1
+
+Private WithEvents mBrandingButton      As RibbonButton
+Attribute mBrandingButton.VB_VarHelpID = -1
+Private WithEvents mStandardButton1     As RibbonButton
+Attribute mStandardButton1.VB_VarHelpID = -1
+Private WithEvents mStandardButton2     As RibbonButton
+Attribute mStandardButton2.VB_VarHelpID = -1
+Private WithEvents mStandardOptions     As RibbonDropDown
+Attribute mStandardOptions.VB_VarHelpID = -1
+Private WithEvents mShowAdvancedToggle  As RibbonToggleButton
+Attribute mShowAdvancedToggle.VB_VarHelpID = -1
+
+Public Function Initialize(ByVal Factory As IRibbonFactory) As RibbonModel
+    On Error GoTo EH
+    Set this.BrandingGroup = Factory.NewRibbonGroup("BrandingGroup")
+    Set mBrandingButton = Factory.NewRibbonButtonMso("BrandingButton", , , , "MacroSecurity", True)
+    
+    Set this.CustomGroup = Factory.NewRibbonGroup("CustomButtonsGroup", False)
+    Set mCustomButton1 = Factory.NewRibbonButtonMso("AppLaunchButton1", True, True, rdLarge, "RefreshAll", True, True)
+    Set mCustomButton2 = Factory.NewRibbonButtonMso("AppLaunchButton2", True, True, rdLarge, "Refresh", True, True)
+    Set mCustomButton3 = Factory.NewRibbonButtonMso("AppLaunchButton3", True, True, rdLarge, "MacroPlay", True, True)
+    Set mCustomSizeToggle = Factory.NewRibbonToggleMso("SizeToggle", True, True, rdLarge, DefaultImage, False, True)
+    Set mCustomOptions = Factory.NewRibbonDropDown("ButtonOptions2", True, False)
+    mCustomOptions.AddItem Factory.NewSelectableItem("LabelOnly")
+    mCustomOptions.AddItem Factory.NewSelectableItem("ImageOnly")
+    mCustomOptions.AddItem Factory.NewSelectableItem("LabelAndImage")
+    
+    Set this.StandardGroup = Factory.NewRibbonGroup("StandardButtonsGroup")
+    Set mStandardButton1 = Factory.NewRibbonButtonMso("StandardButton1", True, True, rdRegular, "RefreshAll")
+    Set mStandardButton2 = Factory.NewRibbonButtonMso("StandardButton2", True, True, rdRegular, "Refresh")
+    Set mStandardOptions = Factory.NewRibbonDropDown("ButtonOptions", True, True)
+    Set mShowAdvancedToggle = Factory.NewRibbonToggle("ShowAdvancedToggle", True, True)
+    
+    mShowAdvancedToggle.OnToggled False
+    mStandardOptions.OnActionDropDown "", 2
+
+    mCustomOptions.SelectedItemId = "LabelAndImage"
+    mCustomOptions.IsEnabled = mCustomSizeToggle.IsPressed
+    
+    Set Initialize = Me
+XT: Exit Function
+EH: ReraiseError Err, ModuleName & "Initialize"
+    Resume XT
+    Resume Next
+End Function
+
+Private Sub mBrandingButton_Clicked()
+    MsgBox "Quack!", vbOKOnly Or vbInformation, "Rubberduck Ribbon Refresh"
```

Comment:
While this bitwise `Or` is semantically correct, it's probably simpler to just use a `+` operator here.

Reply:
I will think on that. Given that in many, many, years of programming VBA I never discovered that "+" could substitute for "Or" here, perhaps "Or" is clearer.

Reply:
FWIW, I'd rather train users to use `Or`. `+` is wrong because it can screw up the bitmask. (Granted, MsgBox is a weird mix of bitmask & flags, but best to be consistent and use `Or` on bitmasks because they are....*bitmasks*, not some numbers to be added).

Reply:
@bclothier bitwise Or and arithmetic addition are identical operarions AFAIK. TBH I'm torn on this one, between showing users the correct/ bitwise way, and going with what most are familiar with. Aa long as we're consistent about it.

Reply:
I'm sure you already know but technically they are similar but *not* identical. A `vbInformation + vbInformation` will yield bogus result whereas `vbInformation Or vbInformation` is still `vbInformation`; no harm done. It's unlikely an user will repeat a bitmask within a single mask but when it's used in a wrapping function to enforce preset defaults, it makes for a big difference.

Reply:
@bclothier ha, never occurred to me someone could do that - that settles it then.